tree-sitter: update grammars

oxalica opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

There are node changes on lua and/or vim parsers, which cause highlight errors with current version of nvim-treesitter in nixpkgs. This update should fix them.

Note that `tree-sitter-toml` is forked under the official organization `tree-sitter` with a few patch commits applied on HEAD. Thus I switch it to the patched one. Tested to work well.

Before this PR:
- Opening a `.vim` file emits an error.
- `:checkhealth nvim-treesitter` shows errors with java, latex and vim parsers.

After this PR:
- `.vim` and inlined lua are highlighted fine.
- `:checkhealth nvim-treesitter` shows no error.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>tree-sitter-grammars.tree-sitter-c-sharp</li>
    <li>tree-sitter-grammars.tree-sitter-clojure</li>
    <li>tree-sitter-grammars.tree-sitter-comment</li>
    <li>tree-sitter-grammars.tree-sitter-devicetree</li>
    <li>tree-sitter-grammars.tree-sitter-elixir</li>
    <li>tree-sitter-grammars.tree-sitter-go</li>
    <li>tree-sitter-grammars.tree-sitter-gomod</li>
    <li>tree-sitter-grammars.tree-sitter-heex</li>
    <li>tree-sitter-grammars.tree-sitter-java</li>
    <li>tree-sitter-grammars.tree-sitter-json</li>
    <li>tree-sitter-grammars.tree-sitter-julia</li>
    <li>tree-sitter-grammars.tree-sitter-latex</li>
    <li>tree-sitter-grammars.tree-sitter-ledger</li>
    <li>tree-sitter-grammars.tree-sitter-llvm</li>
    <li>tree-sitter-grammars.tree-sitter-lua</li>
    <li>tree-sitter-grammars.tree-sitter-markdown</li>
    <li>tree-sitter-grammars.tree-sitter-nix</li>
    <li>tree-sitter-grammars.tree-sitter-org-nvim</li>
    <li>tree-sitter-grammars.tree-sitter-php</li>
    <li>tree-sitter-grammars.tree-sitter-prisma</li>
    <li>tree-sitter-grammars.tree-sitter-r</li>
    <li>tree-sitter-grammars.tree-sitter-scala</li>
    <li>tree-sitter-grammars.tree-sitter-scheme</li>
    <li>tree-sitter-grammars.tree-sitter-supercollider</li>
    <li>tree-sitter-grammars.tree-sitter-svelte</li>
    <li>tree-sitter-grammars.tree-sitter-toml</li>
    <li>tree-sitter-grammars.tree-sitter-tsx</li>
    <li>tree-sitter-grammars.tree-sitter-typescript</li>
    <li>tree-sitter-grammars.tree-sitter-vim</li>
    <li>tree-sitter-grammars.tree-sitter-zig</li>
  </ul>
</details>


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

oxalica committed:
tree-sitter: update grammars

There are node changes on lua and/or vim parsers, which cause highlight
errors with current version of nvim-treesitter in nixpkgs.
This update should fix them.

QuantMint commented:
Result of `nixpkgs-review pr 170202` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>tree-sitter-grammars.tree-sitter-c-sharp</li>
    <li>tree-sitter-grammars.tree-sitter-clojure</li>
    <li>tree-sitter-grammars.tree-sitter-comment</li>
    <li>tree-sitter-grammars.tree-sitter-devicetree</li>
    <li>tree-sitter-grammars.tree-sitter-elixir</li>
    <li>tree-sitter-grammars.tree-sitter-go</li>
    <li>tree-sitter-grammars.tree-sitter-gomod</li>
    <li>tree-sitter-grammars.tree-sitter-heex</li>
    <li>tree-sitter-grammars.tree-sitter-java</li>
    <li>tree-sitter-grammars.tree-sitter-json</li>
    <li>tree-sitter-grammars.tree-sitter-julia</li>
    <li>tree-sitter-grammars.tree-sitter-latex</li>
    <li>tree-sitter-grammars.tree-sitter-ledger</li>
    <li>tree-sitter-grammars.tree-sitter-llvm</li>
    <li>tree-sitter-grammars.tree-sitter-lua</li>
    <li>tree-sitter-grammars.tree-sitter-markdown</li>
    <li>tree-sitter-grammars.tree-sitter-nix</li>
    <li>tree-sitter-grammars.tree-sitter-org-nvim</li>
    <li>tree-sitter-grammars.tree-sitter-php</li>
    <li>tree-sitter-grammars.tree-sitter-prisma</li>
    <li>tree-sitter-grammars.tree-sitter-r</li>
    <li>tree-sitter-grammars.tree-sitter-scala</li>
    <li>tree-sitter-grammars.tree-sitter-scheme</li>
    <li>tree-sitter-grammars.tree-sitter-supercollider</li>
    <li>tree-sitter-grammars.tree-sitter-svelte</li>
    <li>tree-sitter-grammars.tree-sitter-toml</li>
    <li>tree-sitter-grammars.tree-sitter-tsx</li>
    <li>tree-sitter-grammars.tree-sitter-typescript</li>
    <li>tree-sitter-grammars.tree-sitter-vim</li>
    <li>tree-sitter-grammars.tree-sitter-zig</li>
  </ul>
</details>

Kranzes commented:
Have you tested the issue mentioned here? https://github.com/NixOS/nixpkgs/pull/165722

oxalica committed:
tree-sitter: revert the parser for nix

The new version of parser is out of sync with nvim-treesitter's queries
and causes highlighting failures.

oxalica commented:
> Have you tested the issue mentioned here? #165722

Oops, I'm always using [my own fork of `tree-sitter-nix` and queries](https://github.com/oxalica/tree-sitter-nix), since the original author seems not quite available, and haven't realized the breakage.
I reverted its update for now. Maybe we should open a PR to `nvim-treesitter` to update queries first.
